src/backend/distributed/master/master_modify_multiple_shards.c

```diff
@@ -198,10 +198,12 @@ HasReplication(List *shardIntervalList)
  * the shards when necessary before calling SendQueryToShards.
  */
 static int
-SendQueryToShards(Query *query, List *shardIntervalList)
+SendQueryToShards(Query *query, List *shardIntervalList, Oid relationId)
 {
 	int affectedTupleCount = 0;
-	HTAB *shardConnectionHash = OpenConnectionsToAllShardPlacements(shardIntervalList);
+	char *relationOwner = TableOwner(relationId);
+	HTAB *shardConnectionHash = OpenTransactionsToAllShardPlacements(shardIntervalList,
```

Comment:
I just realised that if you move RegisterShardPlacementXactCallback() into OpenTransactionsToAllShardPlacements, the behaviour would change here, but what if we remove the PG_TRY/PG_CATCH block now?